src/main/java/edu/cornell/kfs/coa/document/validation/impl/GlobalIndirectCostRecoveryAccountsRule.java

```diff
@@ -0,0 +1,392 @@
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.coa.businessobject.A21IndirectCostRecoveryAccount;
+import org.kuali.kfs.coa.businessobject.Account;
+import org.kuali.kfs.coa.businessobject.Chart;
+import org.kuali.kfs.coa.businessobject.IndirectCostRecoveryAccount;
+import org.kuali.kfs.coa.document.validation.impl.GlobalDocumentRuleBase;
+import org.kuali.kfs.coa.service.SubFundGroupService;
+import org.kuali.kfs.sys.KFSKeyConstants;
+import org.kuali.kfs.sys.KFSPropertyConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.kns.document.MaintenanceDocument;
+import org.kuali.rice.krad.bo.GlobalBusinessObjectDetailBase;
+import org.kuali.rice.krad.bo.PersistableBusinessObject;
+import org.kuali.rice.krad.util.GlobalVariables;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+import edu.cornell.kfs.coa.businessobject.IndirectCostRecoveryAccountChange;
+import edu.cornell.kfs.coa.businessobject.SubAccountGlobal;
+import edu.cornell.kfs.coa.businessobject.SubAccountGlobalDetail;
+import edu.cornell.kfs.sys.CUKFSKeyConstants;
+import edu.cornell.kfs.sys.CUKFSPropertyConstants;
+
+public class GlobalIndirectCostRecoveryAccountsRule extends GlobalDocumentRuleBase {
+
+	protected static final BigDecimal BD100 = new BigDecimal(100);
+
+	protected SubFundGroupService subFundGroupService;
+
+	/**
+	 * @see org.kuali.rice.kns.maintenance.rules.MaintenanceDocumentRuleBase#processCustomAddCollectionLineBusinessRules(org.kuali.rice.kns.document.MaintenanceDocument, java.lang.String, org.kuali.rice.krad.bo.PersistableBusinessObject)
+	 */
+	@Override
+	public boolean processCustomAddCollectionLineBusinessRules(MaintenanceDocument document, String collectionName, PersistableBusinessObject line) {
+		boolean success = true;
+		success &= super.processCustomAddCollectionLineBusinessRules(document, collectionName, line);
+		line.refreshNonUpdateableReferences();
+
+		if (line instanceof IndirectCostRecoveryAccountChange) {
+			IndirectCostRecoveryAccountChange account = (IndirectCostRecoveryAccountChange) line;
+			success &= checkIndirectCostRecoveryAccount(account);
+		}
+
+		return success;
+	}
+	
+	/**
+	 * @see org.kuali.rice.kns.maintenance.rules.MaintenanceDocumentRuleBase#processCustomSaveDocumentBusinessRules(org.kuali.rice.kns.document.MaintenanceDocument)
+	 */
+	@Override
+	protected boolean processCustomSaveDocumentBusinessRules(MaintenanceDocument document) {
+		boolean success = super.processCustomSaveDocumentBusinessRules(document);
+		SubAccountGlobal subAccountGlobal = (SubAccountGlobal) getNewBo();
+		
+		if( ObjectUtils.isNotNull(subAccountGlobal.getIndirectCostRecoveryAccounts()) && subAccountGlobal.getIndirectCostRecoveryAccounts().size() >0){
+			validateIndirectCostRecoveryAccounts(subAccountGlobal.getIndirectCostRecoveryAccounts());
```

Comment:
Chad, that is correct, the intention here was for the doc to be saved even if errors occurred. Validation errors are still displayed but don't stop the doc from being saved. I checked on the other Global docs and as you mentioned the functionality was similar.